tests/testGncOptimizer.cpp

```diff
@@ -0,0 +1,670 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    testGncOptimizer.cpp
+ * @brief   Unit tests for GncOptimizer class
+ * @author  Jingnan Shi
+ * @author  Luca Carlone
+ * @author  Frank Dellaert
+ *
+ * Implementation of the paper: Yang, Antonante, Tzoumas, Carlone, "Graduated
+ * Non-Convexity for Robust Spatial Perception: From Non-Minimal Solvers to
+ * Global Outlier Rejection", ICRA/RAL, 2020. (arxiv version:
+ * https://arxiv.org/pdf/1909.08605.pdf)
+ *
+ * See also:
+ * Antonante, Tzoumas, Yang, Carlone, "Outlier-Robust Estimation: Hardness,
+ * Minimally-Tuned Algorithms, and Applications", arxiv:
+ * https://arxiv.org/pdf/2007.15109.pdf, 2020.
+ */
+
+#include <CppUnitLite/TestHarness.h>
+#include <gtsam/nonlinear/GncOptimizer.h>
+#include <gtsam/nonlinear/LinearContainerFactor.h>
+#include <gtsam/slam/dataset.h>
+#include <tests/smallExample.h>
+
+using namespace std;
+using namespace gtsam;
+
+using symbol_shorthand::L;
+using symbol_shorthand::X;
+static double tol = 1e-7;
+
+/* ************************************************************************* */
+TEST(GncOptimizer, gncParamsConstructor) {
+  // check params are correctly parsed
+  LevenbergMarquardtParams lmParams;
+  GncParams<LevenbergMarquardtParams> gncParams1(lmParams);
```

Comment:
Do we need the <> here (and below) or can the type be derived from the argument? I think it can and should be omitted.

Reply:
@dellaert : it does not work if I omit - should I do something else to make it derive the type?

Reply:
Hmmm, I had expected that to work, but maybe it's only since C++17: https://en.cppreference.com/w/cpp/language/class_template_argument_deduction